src/js/components/GridItem.js

```diff
@@ -1,147 +1,147 @@
-          }
-          break;
-        case 'album':
-          dispatch(mopidyActions.getImages([item.uri]));
-          break;
-        default:
-          break;
-      }
-    }
-  }, [item.images]);
-
-  // Build link
-  let to = '';
-  if (getLink) {
-    to = getLink(item);
-  } else if (item.link) {
-    to = item.link;
-  } else {
-    to = `/${item.type}/${encodeUri(item.uri)}`;
-  }
-
-  return (
-    <Link
-      to={to}
-      onContextMenu={onContextMenu}
-      className={`grid__item grid__item--${itemProp.type}`}
-    >
-      <Thumbnail
-        glow={grid_glow_enabled}
-        size="medium"
-        className="grid__item__thumbnail"
-        images={item.images || item.icons}
-        type={item.type}
-      />
-      <div className="grid__item__name">
-        {item.name ? item.name : <span className="opaque-text">{item.uri}</span>}
-      </div>
-      <div className="grid__item__secondary">
-        {show_source_icon && (
-          <Icon name={sourceIcon(item.uri)} type="fontawesome" className="source" />
-        )}
-        <SecondaryLine item={item} />
-      </div>
-    </Link>
-  );
-};
-
-export {
-  GridItem,
-};
-
-export default {
-  GridItem,
-};
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { scrollTo, sourceIcon } from '../util/helpers';
+import Link from './Link';
+import Icon from './Icon';
+import Thumbnail from './Thumbnail';
+import LinksSentence from './LinksSentence';
+import { I18n } from '../locale';
+import { encodeUri } from '../util/format';
+
+import * as uiActions from '../services/ui/actions';
+import * as mopidyActions from '../services/mopidy/actions';
+import * as spotifyActions from '../services/spotify/actions';
+
+const SecondaryLine = ({
+  item: {
+    type,
+    tracks_total,
+    tracks = [],
+    followers,
+    albums_uris,
+    artists,
+  } = {},
+}) => {
+  let trackCount = 0;
+  if (tracks) trackCount = tracks.length;
+  if (tracks_total) trackCount = tracks_total;
+
+  switch (type) {
+    case 'playlist':
+      return (
+        <ul className="grid__item__secondary__content details">
+          <li><I18n path="specs.tracks" count={trackCount} /></li>
+        </ul>
+      );
+
+    case 'artist':
+      return (
+        <ul className="grid__item__secondary__content details">
+          {followers && <li><I18n path="specs.followers" count={followers.toLocaleString()} /></li>}
+          {albums_uris && <li><I18n path="specs.albums" count={albums_uris.length} /></li>}
+        </ul>
+      );
+
+    case 'album':
+      return (
+        <ul className="grid__item__secondary__content details">
+          <li>{artists && <LinksSentence nolinks items={artists} type="artist" />}</li>
+        </ul>
+      );
+
+    default:
+      return (
+        <ul className="grid__item__secondary__content details">
+          {artists && <li><LinksSentence nolinks items={artists} type="artist" /></li> }
```

Comment:
This is the line changed by @viaregio, however the merge resolution recognized the whole file as being changed. I have merged this anyway to appropriately credit Ulrich for his contribution.